sympy/concrete/guess.py

```diff
@@ -377,3 +378,60 @@ def guess_generating_function(v, X=Symbol('x'), types=['all'], maxsqrtn=2):
                 break
 
     return result
+
+
+@public
+def guess(l, stop=False, evaluate=True, niter=None):
+    """
+    This function is adapted from the Rate.m package for Mathematica
+    written by Christian Krattenthaler.
+    It tries to guess a formula from a given sequence of rational numbers.
+
+    In order to speed up the process, the 'stop' variable allows to stop
```

Comment:
The two results are
```
In [5]: print(guess([1,2,6,24,120,720], evaluate=False, stop=False))
[Product(i1 + 1, (i1, 1, i0 - 1)), Product(2*Product((i2 + 2)/(i2 + 1), (i2, 1, i1 - 1)), (i1, 1, i0 - 1))]
```
but the second result is the same as the first one in a more convoluted form since
```
In [12]: print(simplify(2*Product((i2 + 2)/(i2 + 1), (i2, 1, i1 - 1)).doit()))
i1 + 1
```

I suspect that this is what will generally happen if several results are obtained. Therefore I think that looping should stop by default once some results have been obtained (i.e., r is not empty). The name of the flag could perhaps be `all` instead of `stop`, and that could have the default value `False`.